cli/src/program.rs

```diff
@@ -4255,6 +4418,46 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_cli_parse_migrate_program() {
```

Comment:
Should we test the error paths too?

Reply:
This is the CLI. Error handling here is only for slightly nicer error messages. The actual security comes from what we test in `test_loader_v3_to_v4_migration`.